### PR TITLE
2725 recruitment admin tool to update organizations images

### PR DESCRIPTION
**Pull request by superhvarn**
## Changes

- Added two upload buttons in the admin tools
- Download and display capabilities
- hook for setting images
- Added applyInterest and exploreAsGuest to organization shared type
- Testing with test google drive
- Edited set images service method to handle null values

## Screenshots
Screenshot 2024-10-02 at 11 46 36 PM

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes # (issue #2725 )

### CODE REVIEW

**Review by Aaryan1203 — CHANGES_REQUESTED** [review]
Right now when I upload, the loading indicator comes up for both buttons. Also, you can use my NERUplaodButton because with that it gives u a submit and cancel button which would be useful for this as well because right now the user doesn't have any option to not upload the picture when it comes up.

Can you make the two buttons side by side instead of one on top of the other

I get this when I upload smth rn:
image

File: src/frontend/src/pages/HomePage/GuestHomePage.tsx [hunk 1]
```diff
@@ -6,32 +6,48 @@ import { routes } from '../../utils/routes';
...
+          flexDirection: { xs: 'column', md: 'row' },
+          mt: 4,
+          padding: '20px'
+        }}
+      >
+        <Box sx={{ display: 'flex', gap: 5 }}>
           <ImageWithButton
             title="Interested in applying"
-            imageSrc={`/Apply.png`}
+            imageSrc={`https://drive.google.com/thumbnail?id=${organization?.applyInterestImageId}&sz=w1000`}
```
Comment (Aaryan1203) [inline src/frontend/src/pages/HomePage/GuestHomePage.tsx]: use the useGetimageUrl hook. take a look at how I did it in the team type section

**Review by Aaryan1203 — CHANGES_REQUESTED** [review]

File: src/backend/src/services/organizations.services.ts
Comment (Aaryan1203) [inline src/backend/src/services/organizations.services.ts:1]: There is an endpoint in this file called getOrganizationImages which is not beiung used. Can you delete that from the routes controller and service

**Review by Aaryan1203 — CHANGES_REQUESTED** [review]
Can you put the two upload buttons next to eachother instead of one on top of the other.

Also, when you upload an image, there should be a loading indicator before it pops up.

When i upload an image right now, I don't see it unless I refresh the page.

File: src/frontend/src/pages/AdminToolsPage/RecruitmentConfig/AdminToolsRecruitmentConfig.tsx [hunk 2]
```diff
@@ -1,12 +1,55 @@
...
-import ApplicationLinkTable from './ApplicationLinkTable';
+import { useToast } from '../../../hooks/toasts.hooks';
+import NERUploadButton from '../../../components/NERUploadButton';
+import React, { useState } from 'react';
+import { useCurrentOrganization, useSetOrganizationImages } from '../../../hooks/organizations.hooks';
+import LoadingIndicator from '../../../components/LoadingIndicator';
+import { useGetImageUrl } from '../../../hooks/onboarding.hooks';
 
 const AdminToolsRecruitmentConfig: React.FC = () => {
+  const { mutateAsync: organizationImages } = useSetOrganizationImages();
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/AdminToolsPage/RecruitmentConfig/AdminToolsRecruitmentConfig.tsx]: add isLoading check here

File: src/frontend/src/pages/AdminToolsPage/RecruitmentConfig/AdminToolsRecruitmentConfig.tsx [hunk 3]
```diff
@@ -1,12 +1,55 @@
...
+  const toast = useToast();
+  const { data: organization } = useCurrentOrganization();
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/AdminToolsPage/RecruitmentConfig/AdminToolsRecruitmentConfig.tsx]: add isLoading, isError and error to all of these and associated checks

File: src/frontend/src/pages/HomePage/GuestHomePage.tsx [hunk 4]
```diff
@@ -6,32 +6,63 @@ import { routes } from '../../utils/routes';
...
+    isError: exploreImageError
+  } = useGetImageUrl(organization?.exploreAsGuestImageId ?? null);
+
   useEffect(() => {
     setCurrentHomePage('guest');
   }, [setCurrentHomePage]);
 
+  if (!organization || isLoading || applyImageLoading || exploreImageLoading) return <LoadingIndicator />;
+  if (isError) return <ErrorPage message={error?.message} />;
+  if (applyImageError) return <ErrorPage message="Error loading apply interest image" />;
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/HomePage/GuestHomePage.tsx]: include the individual error messages here as well

File: src/frontend/src/pages/HomePage/GuestHomePage.tsx [hunk 5]
```diff
@@ -6,32 +6,63 @@ import { routes } from '../../utils/routes';
...
+  if (exploreImageError) return <ErrorPage message="Error loading explore as guest image" />;
+  if (!applyInterestImageUrl || !exploreGuestImageUrl) throw new Error('Image URLs are undefined');
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/HomePage/GuestHomePage.tsx]: woah were not throwing errors in componenets, we should be returning the loading indivator

File: src/frontend/src/pages/HomePage/GuestHomePage.tsx [hunk 6]
```diff
@@ -6,32 +6,63 @@ import { routes } from '../../utils/routes';
...
+    data: exploreGuestImageUrl,
+    isLoading: exploreImageLoading,
+    isError: exploreImageError
+  } = useGetImageUrl(organization?.exploreAsGuestImageId ?? null);
+
   useEffect(() => {
     setCurrentHomePage('guest');
   }, [setCurrentHomePage]);
 
+  if (!organization || isLoading || applyImageLoading || exploreImageLoading) return <LoadingIndicator />;
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/HomePage/GuestHomePage.tsx]: return loading indicator after errors

**Review by Peyton-McKee — CHANGES_REQUESTED** [review]
should be done after these changes

**Pull request merged**